examples/bun-yoga-ws/__integration-tests__/bun.spec.ts

```diff
@@ -0,0 +1,71 @@
+import { Server } from 'bun';
+import { describe, expect, it } from 'bun:test';
+import { createSchema, createYoga } from 'graphql-yoga';
+
+describe('Bun integration', () => {
+  const yoga = createYoga({
+    schema: createSchema({
+      typeDefs: /* GraphQL */ `
+        type Query {
+          greetings: String
+        }
+      `,
+      resolvers: {
+        Query: {
+          greetings: () => 'Hello Bun!',
+        },
+      },
+    }),
+  });
+
+  let server: Server;
+  let url: string;
+  function beforeEach() {
+    server = Bun.serve({
+      fetch: yoga,
+      port: 3000,
+    });
+    url = `http://${server.hostname}:${server.port}${yoga.graphqlEndpoint}`;
+  }
+
+  function afterEach() {
+    server.stop();
+  }
+
+  it('shows GraphiQL', async () => {
+    beforeEach();
```

Comment:
Why not using the `beforeEach` and `afterEach` lifecycle hooks instead of manually calling setup and tear down functions ?

Reply:
When I wrote this test, Bun didn't have those hooks :/ Not sure if it has now.

Reply:
Loookig at the documentation, it should :-) I don't know when they added them